src/views/Patient/MedReqDropDown/cdsHooksCards/cdsHooksCard.tsx

```diff
@@ -16,148 +17,183 @@ import { Card as HooksCard, Link } from '../../../../cds-hooks/resources/HookTyp
 //  - look into using the fhir client directly instead of using axios
 
 interface CdsHooksCardProps {
-    card: HooksCard
-    client: Client
+  card: HooksCard;
+  client: Client;
+  tabCallback: (n: ReactElement, m: string) => void;
 }
 
 const CdsHooksCard = (props: CdsHooksCardProps) => {
-
-    const [links, setLinks] = useState<Link[]>([]);
-
-    useEffect(() => {
-        modifySmartLaunchURLs(props.card).then((updatedLinks) => {
-            setLinks(updatedLinks);
-            console.log('CdsHooksCard::useEffect: updated all of the smart links for: ' + props.card?.summary);
-        });
-
-    }, [props.card]);
-
-    function retrieveLaunchContext(client: Client, link: Link) {
-
-        const patientId = client?.patient?.id;
-        const accessToken = client?.state?.tokenResponse?.access_token;
-        const fhirBaseUrl = client?.state?.serverUrl;
-
-        return new Promise<Link>((resolve, reject) => {
-            const headers = accessToken ?
-            {
-                'Accept': 'application/json',
-                'Authorization': `Bearer ${accessToken}`
-            }
-            :
-            {        
-                'Accept': 'application/json'
-            };
-            const launchParameters = {
-                patient: patientId,
-                appContext: ''
-            };
-        
-            if (link.appContext) {
-                launchParameters.appContext = link.appContext;
+  const [links, setLinks] = useState<Link[]>([]);
+
+  useEffect(() => {
+    modifySmartLaunchURLs(props.card).then(updatedLinks => {
+      setLinks(updatedLinks);
+      console.log(
+        'CdsHooksCard::useEffect: updated all of the smart links for: ' + props.card?.summary
+      );
+    });
+  }, [props.card]);
+
+  function retrieveLaunchContext(client: Client, link: Link) {
```

Comment:
Will the smart app only work to launch the built-in DTR? Or can the hook point to a different smart app?

Reply:
This looks and works great, after my other questions are addressed, we just need to merge the latest from dev. Differences to merge can be found here:
https://github.com/mcode/rems-smart-on-fhir/pull/23/files

Reply:
I would've said that it should be able to point to a different SMART app, to satisfy CRD requirements, but I actually think having it only launch the built-in DTR is more forward thinking, given upcoming changes to the CRD IG where it will no longer provide a link to a SMART app.